src/backend/distributed/commands/utility_hook.c

```diff
@@ -468,6 +468,18 @@ ProcessUtilityInternal(PlannedStmt *pstmt,
 		PreprocessTruncateStatement((TruncateStmt *) parsetree);
 	}
 
+	if (IsA(parsetree, LockStmt))
+	{
+		LockStmt *stmt = (LockStmt *) parsetree;
```

Comment:
Is there anything that prevents us from moving this into a preprocess function? (at least similar to what we have for truncate?)

Reply:
The code would not be executed if we move it to a preprocess function due to preprocess being executed only for `EnableDDLPropagation` set to true. I would expect nodes with `EnableDDLPropagation` set to false to be able to lock relations whose shards they still own.

Reply:
Not sure what the right behaviour is for enable_ddl_propagation, but in any case we should be able to move this code into a PreprocessLockStatement function, analogous to PreprocessTruncateStatement.